frontend/src/host_orchestrator/orchestrator/controller.go

```diff
@@ -209,3 +213,24 @@ type listUploadDirectoriesHandler struct {
 func (h *listUploadDirectoriesHandler) Handle(r *http.Request) (interface{}, error) {
 	return h.m.ListDirs()
 }
+
+type createUpdateUserArtifactHandler struct {
+	m UserArtifactsManager
+}
+
+func (h *createUpdateUserArtifactHandler) Handle(r *http.Request) (interface{}, error) {
+	vars := mux.Vars(r)
+	dir := vars["name"]
+	f, fheader, err := r.FormFile("file")
+	if err != nil {
+		if err == multipart.ErrMessageTooLarge {
+			return nil, &operator.AppError{
+				StatusCode: http.StatusInsufficientStorage,
+				Err:        err,
+			}
+		}
+		return nil, operator.NewBadRequestError("Invalid multipart form request", err)
+	}
+	defer r.MultipartForm.RemoveAll()
+	return nil, h.m.CreateUpdateArtifact(dir, fheader.Filename, f)
```

Comment:
Is it safe to use `fheader.Filename`? For example, is it guaranteed to not be something like "../../../usr/..."?

Reply:
Specs required that the path information should be stripped and Go complies with it: https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/mime/multipart/multipart.go;l=84;drc=7c7cd56870ba617f964014fa4694e9b61e29cf97